File: LABS/LAB_03/src/ki304/vasylkiv/lab3/ActivityStatus.java
package ki304.vasylkiv.lab3;


/**
 * Class <code>ActivityStatus</code> implements the status of a person's activity
 */
public class ActivityStatus {

    enum Activities {
        MOVING, STANDING, SITTING, LIE
    }
   
    private Activities activity;
    
    /**
     * Constructor
     */
    public ActivityStatus() {
    	activity = Activities.STANDING;
    }
    
    /**
     * Method changes the activity mode to MOVING
     */
    public void setMovingActivity() {
    	activity = Activities.MOVING;
    }
    
    /**
     * Method changes the activity mode to STANDING
     */
    public void setStangingActivity() {
    	activity = Activities.STANDING;
    }
    
    /**
     * Method changes the activity mode to SITTING
     */
    public void setSittingActivity() {
    	activity = Activities.SITTING;
    }
    
    /**
     * Method changes the activity mode to LIE
     */
    public void setLieActivity() {
    	activity = Activities.LIE;
    }
    
    /**
     * Return Current Activity
     * 
     * @return Activities
     */
    public Activities getCurrentActivity() {
        return activity;
    }
  
}

File: LABS/LAB_09/ActivityStatus.py
from enum import Enum

class ActivityStatus:
   
    class Activities(Enum):
        MOVING = 1
        STANDING = 2
        SITTING = 3
        LIE = 4

    # Constructor
    def __init__(self):
        self.__activity = self.Activities.STANDING

    # Method: Sets the activity to MOVING.
    def setMovingActivity(self):
        self.__activity = self.Activities.MOVING

    # Method: Sets the activity to STANDING.
    def setStandingActivity(self):
        self.__activity = self.Activities.STANDING

    # Method: Sets the activity to SITTING.
    def setSittingActivity(self):
        self.__activity = self.Activities.SITTING

    # Method: Sets the activity to LIE.
    def setLieActivity(self):
        self.__activity = self.Activities.LIE

    # Method: Returns the current activity mode as a string.
    def getCurrentActivity(self):
        return self.__activity.name
